hi-ml/src/health_ml/model_trainer.py

```diff
@@ -0,0 +1,199 @@
+                      logger=loggers,
+                      progress_bar_refresh_rate=progress_bar_refresh_rate,
+                      num_nodes=num_nodes,
+                      gpus=num_gpus,
+                      precision=precision,
+                      sync_batchnorm=True,
+                      terminate_on_nan=container.detect_anomaly,
+                      profiler=container.pl_profiler,
+                      resume_from_checkpoint=str(resume_from_checkpoint) if resume_from_checkpoint else None,
+                      **kwargs)
+    return trainer, storing_logger
+
+
+def model_train(container: LightningContainer,
+                num_nodes: int = 1) -> Tuple[Trainer, StoringLogger]:
+    """
+    The main training loop. It creates the Pytorch model based on the configuration options passed in,
+    creates a Pytorch Lightning trainer, and trains the model.
+    If a checkpoint was specified, then it loads the checkpoint before resuming training.
+
+    :param num_nodes: The number of nodes to use in distributed training.
+    :param container: A container object that holds the training data in PyTorch Lightning format
+    and the model to train.
+    :return: A tuple of [Trainer, StoringLogger]. Trainer is the Lightning Trainer object that was used for fitting
+    the model. The StoringLogger object is returned when training a built-in model, this is None when
+    fitting other models.
+    """
+    lightning_model = container.model
+
+    # resource_monitor: Optional[ResourceMonitor] = None
+    # Execute some bookkeeping tasks only once if running distributed:
+    if is_global_rank_zero():
+        logging.info(f"Model checkpoints are saved at {container.checkpoint_folder}")
+        write_args_file(container,
+                        outputs_folder=container.outputs_folder)
+
+    data_module = container.get_data_module()
+    if is_global_rank_zero():
+        container.before_training_on_global_rank_zero()
+    if is_local_rank_zero():
+        container.before_training_on_local_rank_zero()
+    container.before_training_on_all_ranks()
+
+    # Create the trainer object. Backup the environment variables before doing that, in case we need to run a second
+    # training in the unit tests.
+    old_environ = dict(os.environ)
+    # Set random seeds just before training
+    seed_everything(container.get_effective_random_seed())
+    trainer, storing_logger = create_lightning_trainer(container,
+                                                       num_nodes=num_nodes,
+                                                       **container.get_trainer_arguments())
+    rank_info = ", ".join(f"{env}: {os.getenv(env)}"
+                          for env in [ENV_GLOBAL_RANK, ENV_LOCAL_RANK, ENV_NODE_RANK])
+    logging.info(f"Environment variables: {rank_info}. trainer.global_rank: {trainer.global_rank}")
+
+    logging.info("Starting training")
+    trainer.fit(lightning_model, datamodule=data_module)
```

Comment:
just saw today that with PL 1.5, the recovery checkpoint shoudl be passed into trainer.fit, rather than Trainer(...)

Reply:
removed this for now since we don't have checkpoint handling yet in this PR